pkg/controllers/user/logging/utils/kafka.go

```diff
@@ -49,91 +56,186 @@ func (w *kafkaTestWrap) TestReachable(dial dialer.Dialer, includeSendTestLog boo
 		return nil
 	}
 
-	for _, v := range w.BrokerEndpoints {
-		url, err := url.Parse(v)
-		if err != nil {
-			return errors.Wrapf(err, "couldn't parse url %s", v)
+	if len(w.BrokerEndpoints) == 0 {
+		return errors.New("broker endpoint is empty")
+	}
+
+	broker0URL, err := url.Parse(w.BrokerEndpoints[0])
+	if err != nil {
+		return errors.Wrapf(err, "couldn't parse url %s", w.BrokerEndpoints[0])
+	}
+
+	var broker0TLSConfig *tls.Config
+	if broker0URL.Scheme == "https" {
+		if broker0TLSConfig, err = buildTLSConfig(w.Certificate, w.ClientCert, w.ClientKey, "", "", broker0URL.Hostname(), true); err != nil {
+			return err
 		}
+	}
 
-		var tlsConfig *tls.Config
-		if url.Scheme == "https" {
-			tlsConfig, err = buildTLSConfig(w.Certificate, w.ClientCert, w.ClientKey, "", "", url.Hostname(), true)
-			if err != nil {
+	broker0Host := broker0URL.Host
+	broker0Conn, err := w.kafkaConn(dial, broker0TLSConfig, broker0Host)
+	if err != nil {
+		return err
+	}
+	defer broker0Conn.Close()
+
+	brokers, err := broker0Conn.Brokers()
+	if err != nil {
+		return errors.Wrap(err, "couldn't get broker list")
+	}
+
+	topicConf := kafka.TopicConfig{
+		Topic:             w.Topic,
+		NumPartitions:     len(brokers),
+		ReplicationFactor: 1,
+	}
+
+	brokerController, err := broker0Conn.Controller()
+	if err != nil {
+		return errors.Wrap(err, "couldn't get broker controller")
+	}
+
+	brokerControllerHost := brokerHost(brokerController.Host, brokerController.Port)
+	if brokerControllerHost == broker0Host {
+		if err := broker0Conn.CreateTopics(topicConf); err != nil {
+			return errors.Wrapf(err, "couldn't create topic %s", w.Topic)
+		}
+
+		if includeSendTestLog {
+			if err = w.sendData2Kafka(broker0Conn, dial); err != nil {
 				return err
 			}
 		}
+	} else {
+		brokerControllerTLSConfig, err := buildTLSConfig(w.Certificate, w.ClientCert, w.ClientKey, "", "", brokerController.Host, true)
```

Comment:
Should we only build TLSConfig when  the schema matches `https` like L69 and formerly?